### PR TITLE
Updating combobx explainer with revised anatomy

### PR DESCRIPTION
**Pull request by sudheer-gowrigari**
Updating `combobx` explainer with revised anatomy as `selectlist` element is not an active proposal currently.

### CODE REVIEW

**Comment by gregwhitworth** [comment]
After a sync with @sudheer-gowrigari, he will update this PR after we revisit: https://github.com/openui/open-ui/issues/939#issuecomment-1910837275

File: site/src/pages/components/combobox.explainer.mdx [hunk 1]
```diff
@@ -46,66 +47,33 @@ Further highlighting the need for such a solution, a recent [Twitter poll](https
...
-    - `<option>`
-    - `<optgroup>` (optional)
-- `<option>`: Represents an individual selectable item within the `listbox`.
-- `<optgroup>` (Optional): A container to group related options together.
-  - Slots:
-    - `<legend>`: To label the group.
-    - `<option>`: The options within this group.
-- `<legend>` (Optional, within `<optgroup>`): Provides a label or title for a group of options within the `listbox`.
+- `<combobox>`: The root container that encapsulates the entire `combobox` structure. It provides context for the interaction between the `input` and the `datalist`.
+- `<input type=text>`: This is where the user can type input or view the selected option. It acts as the trigger to display the datalist.
```
Comment (gregwhitworth) [inline site/src/pages/components/combobox.explainer.mdx:71]: > ... view the selected option

This isn't true per your anatomy. This is what `<selectedoption>` is for. I think a separate issue may be warranted that opens the can of worms that is "input taking complex content".

File: site/src/pages/components/combobox.explainer.mdx [hunk 2]
```diff
@@ -142,80 +110,15 @@ To showcase the selected option within a `combobox`, you can include a `<selecte
             <option>Strawberry</option>
             <option>Blueberry</option>
         </optgroup>
-    </listbox>
+    </datalist>
 </combobox>
 ```
 
-## Introducing `filter` attribute
```
Comment (gregwhitworth) [inline site/src/pages/components/combobox.explainer.mdx:149]: Why did you remove this? Let's include the new attributes and then determine how they want them to be incubated?

**Review by gregwhitworth — CHANGES_REQUESTED** [review]

File: site/src/pages/components/combobox.explainer.mdx (hunk 2 above)
Comment (sudheer-gowrigari) [reply site/src/pages/components/combobox.explainer.mdx]: @gregwhitworth Added back.

**Review by gregwhitworth — APPROVED** [review]
